packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -215,7 +211,8 @@ export function buildCart(engine: CommerceEngine, props: CartProps = {}): Cart {
     },
 
     purchase(transaction: Transaction) {
-      dispatch(purchase(transaction));
+      dispatch(emitPurchase(transaction));
+      dispatch(purchase());
```

Comment:
I'm remembered that I had a discussion about this [guideline](https://redux.js.org/style-guide/#avoid-dispatching-many-actions-sequentially) with Louis. I _think_ it doesn't apply in that situation (because only one action will have an impact on the state), but I wanted to mention it to make sure we are ok with this.

Reply:
Indeed the guideline does not apply in this situation because the emit action doesn't affect the state.